File: app/Entry.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Entry extends Model
{
    protected $fillable = [
        'uid', 'lecture_id',
    ];

    public function user() {
      return $this->belongsTo('App\User', 'uid', 'uid');
    }

    public function lecture() {
      return $this->belongsTo('App\Lecture');
    }
}

File: app/models/entry.rb
# Model class for the +entries+ table
class Entry < ApplicationRecord
  belongs_to :lecture, optional: true
  validates :uid, presence: true, length: { is: 64 }

  # Returns the user object for this entry's uid
  def user
    User.find_by(uid: uid) || NullUser.new
  end
end
